### ISSUE TITLE
Xcode ignores custom rule warning in some cases

### ISSUE THREAD

**Issue by yaroslavyaroslav** [realm/SwiftLint#3757]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
I'm trying to make custom rule that throw warning on any commented code.

So my regex in the `swiftlint.yml` are `(^\\/\\/\\s*(\\.?[a-z]))`. It should match any lint that starts with the `//` and continues with the any `lower_case_letter` either `.lower_case_letter` after optional sequence of whitespaces.

The problem comes with the following code:

```swift
struct Some: View {
    var body: some View {
        VStack(alignment: .leading) {
            EmptyView()
        }
        .navigationBarTitleDisplayMode(.inline)
        .navigationTitle("Title")
//        .navigationBarBackButtonHidden(true) // this line didn't match
    }

}

//struct Some_Previews: PreviewProvider {
//    static var previews: some View { // this line match
//        Some()
//    }
//}
```
So the line with commented out `.navigationBarBackButtonHidden(true)` don't shows after building by Xcode 13.2 beta 1. But it shows in bash output. Also Xcode shows warning for first two lines in preview block (which is expected).

The Xcode starts working well if i make exact and non optional way, like:
- `(^\\/\\/\\s*(\\.?[a-z]))` - don't show warning.
- `(^\\/\\/\\s*(\\.?n))` - shows warning.
- `(^\\/\\/\\s*(\\.?([a-z]|n)))` - don't show warning.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint

Linting Swift files in current working directory
Linting 'UITestUtilityDataModel.swift' (1/29)
Linting 'Contents.swift' (2/29)
Linting 'TrafficRulesExamUITestsLaunchTests.swift' (3/29)
Linting 'TrafficRulesExamTests.swift' (4/29)
Linting 'TrafficRulesExamUITests.swift' (5/29)
Linting 'MainScreen.swift' (6/29)
Linting 'TrafficRulesExamApp.swift' (7/29)
Linting 'TotalStats.swift' (8/29)
/TrafficRulesExamUITests.swift:176:12: warning: Todo Violation: FIXMEs should be resolved (В некоторых вопросах только 2 ...). (todo)
Linting 'ExamCardStats.swift' (9/29)
/path/to/project/Views/MainScreen.swift:50:16: warning: Todo Violation: FIXMEs should be resolved (Переписать из двух экранов вые...). (todo)
Linting 'TotalStatsNavigation.swift' (10/29)
/path/to/project/Views/TrafficRulesExamApp.swift:13:1: warning: Commented out code Violation: No commented code in devel branch allowed. (commented_out_code)
/path/to/project/Views/TrafficRulesExamApp.swift:16:1: warning: Commented out code Violation: No commented code in devel branch allowed. (commented_out_code)
/path/to/project/Views/TrafficRulesExamApp.swift:23:1: warning: Commented out code Violation: No commented code in devel branch allowed. (commented_out_code)
/path/to/project/Views/TrafficRulesExamApp.swift:24:1: warning: Commented out code Violation: No commented code in devel branch allowed. (commented_out_code)
Linting 'ExamStats.swift' (11/29)
Linting 'Card.swift' (12/29)
Linting 'Answers.swift' (13/29)
Linting 'AnswerButton.swift' (14/29)
Linting 'QuestionContent.swift' (15/29)
Linting 'QuestionCard.swift' (16/29)
Linting 'CardItem.swift' (17/29)
Linting 'CardsGalery.swift' (18/29)
Linting 'CardRow.swift' (19/29)
Linting 'RawRepresentable.swift' (20/29)
Linting 'Results.swift' (21/29)
Linting 'Date+PrittyPrint.swift' (22/29)
Linting 'CardResults.swift' (23/29)
Linting 'ExamCard.swift' (24/29)
/path/to/project/Views/Exam/Card.swift:28:3: warning: Comment Spacing Violation: Prefer at least one space after slashes for comments. (comment_spacing)
/path/to/project/Views/Exam/Card.swift:32:3: warning: Comment Spacing Violation: Prefer at least one space after slashes for comments. (comment_spacing)
/path/to/project/Views/Exam/Card.swift:23:1: warning: Commented out code Violation: No commented code in devel branch allowed. (commented_out_code)
/path/to/project/Views/Exam/Card.swift:28:1: warning: Commented out code Violation: No commented code in devel branch allowed. (commented_out_code)
/path/to/project/Views/Exam/Card.swift:29:1: warning: Commented out code Violation: No commented code in devel branch allowed. (commented_out_code)
Linting 'URLSession+async.swift' (25/29)
Linting 'Answer.swift' (26/29)
Linting 'ModelLoad.swift' (27/29)
Linting 'Question.swift' (28/29)
Linting 'UserDefaults.swift' (29/29)
/path/to/project/Model/CardResults.swift:39:16: warning: Todo Violation: TODOs should be resolved (Make something if data nil). (todo)
/path/to/project/Model/ExamCard.swift:24:12: warning: Todo Violation: FIXMEs should be resolved (Remove force unwrap.). (todo)
Done linting! Found 13 violations, 0 serious in 29 files.
```

`/path/to/project/Views/Exam/Card.swift:23:1: warning: Commented out code Violation: No commented code in devel branch allowed. (commented_out_code)` — this is the `//        .navigationBarBackButtonHidden(true) // this line didn't match` in the source file.

### Environment

* 0.45.0
* mint
* Paste your configuration file:

```yml
excluded:
  - Carthage
  - Pods
  - .build
  - Build
  - DerivedData

disabled_rules:
  - type_name
  - identifier_name
  - line_length
  - multiple_closures_with_trailing_closure

# force warnings
force_cast: error
force_try: error 
# regex: "(^\\/\\/\\s*.*)" # matching pattern
custom_rules:
  commented_out_code: # rule identifier
    included: ".*\\.swift" # regex that defines paths to include during linting. optional.
    excluded: ".*Test(s)?\\.swift" # regex that defines paths to exclude during linting. optional
    name: "Commented out code" # rule name. optional.
    regex: "(^\\/\\/\\s*(\\.?([a-z])))" # matching pattern
    capture_group: 0 # number of regex capture group to highlight the rule violation at. optional.
    match_kinds: # SyntaxKinds to match. optional.
      - comment
    message: "No commented code in devel branch allowed." # violation message. optional.
    severity: warning # violation severity. optional.

  force_unwrap_function: # rule identifier
    included: ".*\\.swift" # regex that defines paths to include during linting. optional.
    excluded: ".*Test\\.swift" # regex that defines paths to exclude during linting. optional
    name: "Force unwrap function" # rule name. optional.
    regex: "(\\}\\s*!)" # matching pattern
    capture_group: 0 # number of regex capture group to highlight the rule violation at. optional.
    match_kinds: # SyntaxKinds to match. optional.
      - comment
      - identifier
    message: "Please make it safe way, or disable this warning localy." # violation message. optional.
    severity: error # violation severity. optional.
```

* Which Xcode version are you using (check `xcodebuild -version`)?
   * Xcode 13.2
   * Build version 13C5066c

```swift
// This triggers a violation:
//struct Some_Previews: PreviewProvider {
//    static var previews: some View { // this line match
//        .navigationBarBackButtonHidden(true) // this line didn't match
```

And here's the screen of the issue.
<img width="1340" alt="Снимок экрана 2021-11-17 в 13 46 50" src="https://user-images.githubusercontent.com/16612247/142186653-a47d0c22-9d5d-401c-afb4-0f4ef3b8856a.png">
